### DISCUSSION TITLE
Map certificate expiry notification to an email included in a custom attribute

### DISCUSSION THREAD

**Discussion by ghost** [OmniTrustILM/ilm#47] [Q&A] [answered]
Hello,
is it possible to send certificate expiry notification to an email included in a custom attribute in a specific certificate object? And if so, how is it achievable?

Use case would be a required custom attribute designed for emails, so the user would have to specify an email for notifications related to the specific certificate.

**Comment by 3keyroman** [comment, marked as answer]
No, it is not possible in the current implementation. Notifications are sent for user or group that is assigned to certificate as recipient. The certificate notification data does not contain any mapped custom attributes.

Mapped custom attributes for notification can be used for user or group. The use case is to map information needed to specific notification channel, that can be for example Slack channel ID, or some other. The Notification Provider can work with such mapped custom attributes, but not on the certificate object level (see for example [NotificationDataCertificateStatusChanged](https://github.com/CZERTAINLY/CZERTAINLY-Interfaces/blob/master/src/main/java/com/czertainly/api/model/connector/notification/data/NotificationDataCertificateStatusChanged.java)).

When there are no recipients (user or group), the notification is not sent.

We are going to improve validation and notification for certificates in platform in next quarter. Our goal is to provide higher flexibility for configuration of when certificates managed by RA profiles should be validated and notified, including the possibility to create escalation contacts and more. The notification is dependent on validation currently, because only validation process triggers the change in the certificate status and thus notification.

Why you would like to have custom attribute with email that will be used for notification (email notification in this case)? Can you use group(s) with email address for your use case?

**Comment by ghost** [comment]
Thank you for answer, I was just asking for information and don't have any suggestions for a new feature.
